async.go

```diff
@@ -67,7 +63,7 @@ func ConnectAsync(addr string, keepAlive time.Duration, logger *zerolog.Logger,
 	}
 
 	if err != nil {
-		return nil, errors.WithContext(err, DIAL)
```

Comment:
Cleaning up errors (not using that dumb custom errors.WithContext function which added no information and was a pain to use